test/Frame.spec.jsx

```diff
@@ -313,6 +313,63 @@ describe('The Frame Component', () => {
     );
   });
 
+  it('should call contentWillUnmount on unmount', done => {
+    div = document.body.appendChild(document.createElement('div'));
+    const willUnmount = sinon.spy();
```

Comment:
I see this is called but this test doesn't check the `callCount` either?

Reply:
So, I just need to check the `callCount` to be 1 ryt?

Reply:
yep

Reply:
I have updated the test, The test is failing, I don't exactly get the reason for this. The implementation of the feature seems to correct and working. But the way I wrote the test is what I guess is wrong. Can you help me here?

Reply:
Ok so i think what we're doing here is actually the wrong thing, the issue is about having a callback when the `<Frame>` component unmounts and not when the contents within the iframe unmount.

So I think what we need is a bit simpler instead of passing the `contentWillUnmount` prop down through to `<Content>` we just need to call it inside `<Frame>`'s `componentWillUnmount` callback and we should probably rename it to `frameWillUnmount` to differentiate it from the other content lifecycle events.

Then you can just test it like this:
```js
  it('should call contentWillUnmount on unmount', done => {
    div = document.body.appendChild(document.createElement('div'));
    const willUnmount = sinon.spy();
    const frame = ReactDOM.render(
      <Frame frameWillUnmount={willUnmount} />,
      div
    );
    ReactDOM.unmountComponentAtNode(ReactDOM.findDOMNode(frame).parentNode);
    expect(willUnmount.callCount).to.equal(1, 'expected 1 willUnmount');
    done();
  });
```

Reply:
Okay. But when tried it in an example, the `contentWillUnmount` was called when I unmounted the Frame.

Reply:
Yeah ok having though on this more I think what you have is useful in two instances of changing the iframe contents as well as unmounting the entire Frame too. 

I honestly have no idea why this isn't firing in test env, if you run `npm run karma:dev` that will allow you to debug the tests within Chrome dev tools.

Reply:
Tried with `npm run karma:dev` same issue. How do I proceed? Skip this test?